README.markdown

```diff
@@ -5567,12 +5567,19 @@ This method was introduced in the `0.5.0rc30` release.
 ngx.escape_uri
 --------------
 
-**syntax:** *newstr = ngx.escape_uri(str)*
+**syntax:** *newstr = ngx.escape_uri(str, not_uri_comp?)*
 
 **context:** *init_by_lua&#42;, init_worker_by_lua&#42;, set_by_lua&#42;, rewrite_by_lua&#42;, access_by_lua&#42;, content_by_lua&#42;, header_filter_by_lua&#42;, body_filter_by_lua&#42;, log_by_lua&#42;, ngx.timer.&#42;, balancer_by_lua&#42;, ssl_certificate_by_lua&#42;, ssl_session_fetch_by_lua&#42;, ssl_session_store_by_lua&#42;*
 
 Escape `str` as a URI component.
 
+Since `v0.10.16rc6`, this function accepts an optional boolean `not_uri_comp` argument. When this argument is `true`, these characters bellow will be escaped.
```

Comment:
Personally speaking, it would be more extendable to use an escape type enumeration than to use a boolean variable. 

Maybe we can also list the escaped  characters range when escaping `str` as a URI component as comparison?

Reply:
Yeah, that was also what I original proposed. The optional `type` argument can be of the integer type. This way we can also support other types like `NGX_ESCAPE_ARGS`, `NGX_ESCAPE_HTML`, and etc, just like nginx's own `ngx_escape_uri()` C function.